cpp/include/cugraph/algorithms.hpp

```diff
@@ -303,6 +303,92 @@ void edge_betweenness_centrality(const raft::handle_t& handle,
                                  vertex_t k               = 0,
                                  vertex_t const* vertices = nullptr);
 
+/**
+ * @brief     Compute betweenness centrality for a graph
+ *
+ * Betweenness centrality for a vertex is the sum of the fraction of
+ * all pairs shortest paths that pass through the vertex.
+ *
+ * The current implementation does not support a weighted graph.
+ *
+ * If @p num_vertices and @p vertices are not specified then this function will compute
+ * exact betweenness (compute betweenness using a traversal from all vertices).
+ *
+ * If @p num_vertices or @p vertices are specified then we will compute approximate
+ * betweenness either by random sampling @p num_vertices as the seeds of the traversal, or
+ * by using the provided @p vertices as the seeds of the traversal.
+ *
+ * If both @p and @num_vertices are specified we will throw an error.
```

Comment:
Should we better use `std::optional<std::variant>>`? Then, the code self documents the expected behavior and this comment is unnecessary.